pkg/core/mempool/mem_pool_test.go

```diff
@@ -722,18 +714,114 @@ func TestMempoolAddWithDataGetData(t *testing.T) {
 	require.True(t, mp.ContainsKey(r7.FallbackTransaction.Hash()))
 	r8 := &payload.P2PNotaryRequest{
-		MainTransaction:     newTx(t, 0),
-		FallbackTransaction: newTx(t, smallNetFee-1),
+		MainTransaction:     newTx(0, &nonce),
+		FallbackTransaction: newTx(smallNetFee-1, &nonce),
 	}
 	require.NoError(t, mp.Add(r8.FallbackTransaction, fs, r4))
 	require.True(t, mp.ContainsKey(r8.FallbackTransaction.Hash()))
 	mp.verifiedTxes = append(mp.verifiedTxes[:len(mp.verifiedTxes)-2], mp.verifiedTxes[len(mp.verifiedTxes)-1])
 	_, ok = mp.TryGetData(r7.FallbackTransaction.Hash())
 	require.False(t, ok)
 }
+
+func newTx(netFee int64, nonce *uint32) *transaction.Transaction {
+	tx := transaction.New([]byte{byte(opcode.RET)}, 0)
+	tx.Signers = []transaction.Signer{{}, {}}
+	tx.NetworkFee = netFee
+	*nonce++
+	tx.Nonce = *nonce
+	return tx
+}
+
+func TestMempoolIterateVerifiedTransactions(t *testing.T) {
+	var (
+		smallNetFee        int64 = 3
+		nonce              uint32
+		r1, r2, r3, r4, r5 *payload.P2PNotaryRequest
+	)
+	fs := &FeerStub{
+		feePerByte:  0,
+		p2pSigExt:   true,
+		blockHeight: 5,
+		balance:     100,
+	}
+	mp := New(10, 1, false, nil)
+
+	checkRequestsOrder := func(orderedRequests []*payload.P2PNotaryRequest) {
+		var pooledRequests []*payload.P2PNotaryRequest
+		mp.IterateVerifiedTransactions(func(tx *transaction.Transaction, data any) bool {
+			d := data.(*payload.P2PNotaryRequest)
+			pooledRequests = append(pooledRequests, d)
+			return true
+		})
+		require.Equal(t, orderedRequests, pooledRequests)
+	}
+
+	r1 = &payload.P2PNotaryRequest{
+		MainTransaction:     newTx(0, &nonce),
+		FallbackTransaction: newTx(smallNetFee, &nonce),
+	}
+	require.NoError(t, mp.Add(r1.FallbackTransaction, fs, r1))
+	checkRequestsOrder([]*payload.P2PNotaryRequest{r1})
+
+	// r2 has higher priority than r1. The resulting mp.verifiedTxes: [r2, r1]
+	r2 = &payload.P2PNotaryRequest{
+		MainTransaction:     newTx(0, &nonce),
+		FallbackTransaction: newTx(smallNetFee+1, &nonce),
+	}
+	require.NoError(t, mp.Add(r2.FallbackTransaction, fs, r2))
+	checkRequestsOrder([]*payload.P2PNotaryRequest{r2, r1})
+
+	// r3 has the same priority as r1. The resulting mp.verifiedTxes: [r2, r1, r3]
+	r3 = &payload.P2PNotaryRequest{
+		MainTransaction:     newTx(0, &nonce),
+		FallbackTransaction: newTx(smallNetFee, &nonce),
+	}
+	require.NoError(t, mp.Add(r3.FallbackTransaction, fs, r3))
+	checkRequestsOrder([]*payload.P2PNotaryRequest{r2, r1, r3})
+
+	// r4 has the same priority as r1. The resulting mp.verifiedTxes: [r2, r1, r3, r4]
+	r4 = &payload.P2PNotaryRequest{
+		MainTransaction:     newTx(0, &nonce),
+		FallbackTransaction: newTx(smallNetFee, &nonce),
+	}
+	require.NoError(t, mp.Add(r4.FallbackTransaction, fs, r4))
+	checkRequestsOrder([]*payload.P2PNotaryRequest{r2, r1, r3, r4})
+
+	checkPooledRequest := func(t *testing.T, r *payload.P2PNotaryRequest, isPooled bool) {
+		cont := true
+		var notaryRequest *payload.P2PNotaryRequest
```

Comment:
Use the same notation for group variables declaration, please. It should be either `:=` or `var`, not both.

Reply:
Fixed.